sdk/search/Azure.Search.Documents/src/Indexes/Models/BlobExtractionMode.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Search.Documents.Indexes.Models
+{
+    /// <summary>
+    /// Defines which parts of a blob will be indexed by the blob storage indexer.
+    /// See <see href="https://docs.microsoft.com/azure/search/search-howto-indexing-azure-blob-storage" /> for details.
+    /// </summary>
+    public readonly partial struct BlobExtractionMode : IEquatable<BlobExtractionMode>
+    {
+        private readonly string _value;
+
+        /// <summary> Determines if two <see cref="BlobExtractionMode"/> values are the same. </summary>
+        public BlobExtractionMode(string value)
+        {
+            _value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        private const string StorageMetadataValue = "storageMetadata";
+        private const string AllMetadataValue = "allMetadata";
+        private const string ContentAndMetadataValue = "contentAndMetadata";
+
+        /// <summary>
+        /// Specifies that only the standard blob properties and user-specified metadata will be indexed.
+        /// See <see href="https://docs.microsoft.com/azure/storage/storage-properties-metadata" /> for details.
+        /// </summary>
+        public static BlobExtractionMode StorageMetadata { get; } = new BlobExtractionMode(StorageMetadataValue);
```

Comment:
We should call this `Metadata` to match Track 2 Storage names

Reply:
I'd prefer the "enum" names match the actual value, which is what it would have to be on JS and Python enums as well. I believe Java for extensible enums has the same design.

Reply:
I agree that I would like those names to stay the same, for consistency with the user experience on the Azure portal, which gives choices of ContentAndMetadata, AllMetadata, and StorageMetadata.  (We can change portal if we must but consistency in naming is important.)